agents/heuristics_resource/fence_heuristics_resource.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/libexec/platform-python -tt
+
+import io
+import re
+import subprocess
+import shlex
+import sys, stat
+import logging
+import atexit
+import time
+sys.path.append("/usr/share/fence")
+from fencing import fail_usage, run_command, fence_action, all_opt
+from fencing import atexit_handler, check_input, process_input, show_docs
+from fencing import run_delay
+
+def heuristics_resource(con, options):
+	# Search the node where the resource is running and determine
+	# the ACT node or not. For SBY node, a delay is generated.
+	# Note that this method always returns FALSE.
+
+	if not "--nodename" in options or options["--nodename"] == "":
+		logging.error("nodename parameter required")
+		return False
+
+	if not "--resource" in options or options["--resource"] == "":
+		logging.error("resource parameter required")
+		return False
+
+	target = options["--nodename"]
+	resource = options["--resource"]
+	promotable = options["--promotable"] in ["", "1"]
+	standby_wait = int(options["--standby-wait"])
+	crm_resource_path = options["--crm-resource-path"]
+	crm_node_path = options["--crm-node-path"]
+
+	(rc, out, err) = run_command(options, "%s --name" % crm_node_path)
+	if rc != 0 or out == None:
+		logging.error("Can not get my nodename. rc=%s, stderr=%s" % (rc, err))
+		return False
+
+	mynodename = out.strip()
+
+	if mynodename == target:
+		logging.info("Skip standby wait due to self-fencing.")
+		return False
+
+	(rc, out, err) = run_command(options, "%s -r %s -W" % (crm_resource_path, resource))
+	if rc != 0 or out == None:
+		logging.error("Command failed. rc=%s, stderr=%s" % (rc, err))
+		return False
+
+	search_str = re.compile(r"\s%s%s$" % (mynodename, '\sMaster' if promotable else ''))
+	for line in out.splitlines():
+		searchres = search_str.search(line.strip())
+		if searchres:
+			logging.info("This node is ACT! Skip standby wait.")
+			return False
+
+	logging.info("Resource %s NOT found on this node" % resource)
+
+	if standby_wait > 0:
+		# The SBY node waits for fencing from the ACT node, and tries to fence
+		# the ACT node on next fencing level waking up from sleep.
+		logging.info("Standby wait %s sec" % standby_wait)
+		time.sleep(standby_wait)
+		
+	return False
+
+
+def define_new_opts():
+	all_opt["nodename"] = {
+		"getopt" : "n:",
+		"longopt" : "nodename",
+		"required" : "1",
+		"help" : "-n, --nodename=[nodename]      Name of node to be fenced",
+		"shortdesc" : "Name of node to be fenced",
+		"default" : "",
+		"order" : 1
+		}
+	all_opt["resource"] = {
+		"getopt" : "r:",
+		"longopt" : "resource",
+		"required" : "1",
+		"help" : "-r, --resource=[resource-id]   ID of the resource that should be running in the ACT node",
```

Comment:
I would document that it does not make sense to specify a cloned or bundled resource unless it is promotable and has only a single master instance.

Reply:
Ok. I documented it.